app/javascript/react/screens/App/Overview/components/Migrations/helpers/getMostRecentVMTasksFromRequests.js

```diff
@@ -12,7 +12,13 @@ const getMostRecentVMTasksFromRequests = (requests, actions) => {
 
   const vmTasksForRequestOfPlan = [];
   vmTasksForRequestOfPlan.push(
-    vm_ids.map(vmId => commonUtilitiesHelper.getMostRecentEntityByCreationDate(groupedByVMId[vmId]))
+    vm_ids.reduce((tasks, id) => {
+      const task = commonUtilitiesHelper.getMostRecentEntityByCreationDate(groupedByVMId[id]);
+      if (task) {
+        return [...tasks, task];
+      }
+      return tasks;
+    }, [])
   );
```

Comment:
How about we start defending a little earlier than here?

After line#6, if we add a guard on `allTasks` like this -

```js
if (allTasks.some(task => task.id)) {
 ...
 ...
 // existing code
} else {
  return [];
}
```

I think that will be the only place we will have to worry about.